### PR TITLE
Add `get_ui_field_behaviour()` method to PostgresHook

### LINKED ISSUE #34799: Airflow postgres connection field schema points to database name
### Apache Airflow version

2.7.1

### What happened

Airflow's postgres connection configuration form has a field called  'schema'  which is misguiding as values mentioned here is used to refer to the database name instead of the schema name. It should be correctly named to 'database' or 'dbname'

### What you think should happen instead

_No response_

### How to reproduce

create a connection on the web UI and choose connection type as postgres.
Have a dag connect to an postgres server with multiple databases
provide the database name in the 'schema' field of the connection form- this would work if nothing else is incorrect in the etl
now change the value in the schema field of the connection form to refer to a schema- this will fail unexpectedly as the schema name field actually points to the database name.

### Operating System

Windows and Linux

### Versions of Apache Airflow Providers

2.71

### Deployment

Docker-Compose

### Deployment details

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by pubudu91**
Fixes #34799 and updates the relevant docs page.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by Taragolis — APPROVED** [review]

File: airflow/providers/postgres/hooks/postgres.py [hunk 1]
```diff
@@ -362,3 +362,12 @@ def get_openlineage_database_dialect(self, connection) -> str:
...
         """Returns current schema. This is usually changed with ``SEARCH_PATH`` parameter."""
         return self.get_first("SELECT CURRENT_SCHEMA;")[0]
+
+    @staticmethod
+    def get_ui_field_behaviour() -> dict[str, Any]:
+        return {
+            "hidden_fields": [],
+            "relabeling": {
+                "schema": "Database",
+            }
```
Comment (Taragolis) [inline airflow/providers/postgres/hooks/postgres.py]: Static checks

```suggestion
            },
```

Comment (pubudu91) [reply airflow/providers/postgres/hooks/postgres.py]: Sorry, my bad. Missed this. Should've run the pre-commit checks before sending in.

**Review by Taragolis — APPROVED** [review]

**Comment by pubudu91** [comment]
@Taragolis Any idea about the test failures in the pipeline? The failures don't look like they're related to the change I did right? :thinking:

**Review by Lee-W — APPROVED** [review]

**Comment by Lee-W** [comment]
> @Taragolis Any idea about the test failures in the pipeline? The failures don't look like they're related to the change I did right? 🤔

I would suggest trying to rebase from the latest main and see how it works. Looks like other PRs do not encounter similar issue

**Review by hussein-awala — APPROVED** [review]

**Comment by hussein-awala** [comment]
For the failed tests, I see them failed in different PRs, I'm trying to fix the issue.

**Comment by hussein-awala** [comment]
Congrats on your first commit :tada:

**Comment by pubudu91** [comment]
Thank you for the help all! :smiley:

**Pull request merged**